expression/builtin_encryption_vec.go

```diff
@@ -255,19 +267,31 @@ func (b *builtinEncodeSig) vecEvalString(input *chunk.Chunk, result *chunk.Colum
 	if err1 != nil {
 		return err1
 	}
+	bufTp := b.args[0].GetType()
+	bufEnc := charset.NewEncoding(bufTp.Charset)
 	defer b.bufAllocator.put(buf1)
 	if err := b.args[1].VecEvalString(b.ctx, input, buf1); err != nil {
 		return err
 	}
+	buf1Tp := b.args[1].GetType()
+	buf1Enc := charset.NewEncoding(buf1Tp.Charset)
 	result.ReserveString(n)
+	var encodedBuf []byte
 	for i := 0; i < n; i++ {
 		if buf.IsNull(i) || buf1.IsNull(i) {
 			result.AppendNull()
 			continue
 		}
-		decodeStr := buf.GetString(i)
-		passwordStr := buf1.GetString(i)
-		dataStr, err := encrypt.SQLEncode(decodeStr, passwordStr)
+
+		decodeStr, err := bufEnc.Encode(encodedBuf, buf.GetBytes(i))
```

Comment:
Using `EncodeString` instead of  `Encode`, then we needn't use `SQLEncodeBuf`, we can use `SQLEncode` directly?

Reply:
Done.